internal/db/copy.go

```diff
@@ -4,15 +4,13 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/gadget-inc/dateilager/internal/pb"
 	"github.com/jackc/pgx/v5"
 )
 
 func CopyAllObjects(ctx context.Context, tx pgx.Tx, source int64, target int64) error {
-	var samePackPatterns bool
-	err := tx.QueryRow(ctx, `
-		SELECT COALESCE((SELECT pack_patterns FROM dl.projects WHERE id = $1), '{}') =
-		       COALESCE((SELECT pack_patterns FROM dl.projects WHERE id = $2), '{}');
-	`, source, target).Scan(&samePackPatterns)
+	samePackPatterns, err := HasSamePackPattern(ctx, tx, source, target)
```

Comment:
Makes sense to pull this out into it's own function, but I don't think we need it to be exported, it can remain private to this package